packages/lake/src/components/DatePicker.tsx

```diff
@@ -0,0 +1,1450 @@
+    },
+    [format, onChange],
+  );
+
+  return (
+    <>
+      <YearMonthSelect
+        monthNames={monthNames}
+        value={monthYear}
+        hideArrows={!desktop}
+        onChange={setMonthYear}
+      />
+
+      <Space height={24} />
+
+      <MonthCalendar
+        month={monthYear.month}
+        year={monthYear.year}
+        value={isNotNullishOrEmpty(value) ? parseDate(value, format) : Option.None()}
+        firstWeekDay={firstWeekDay}
+        weekDayNames={weekDayNames}
+        isSelectable={isSelectable}
+        onChange={handleChange}
+      />
+    </>
+  );
+};
+
+export const DatePicker = ({
+  label,
+  value,
+  error,
+  format,
+  firstWeekDay,
+  monthNames,
+  weekDayNames,
+  isSelectable,
+  onChange,
+}: DatePickerProps) => {
+  const { desktop } = useResponsive(DATE_PICKER_MOBILE_THRESHOLD);
+  const ref = useRef<TextInput>(null);
+  const [isOpened, { open, close }] = useDisclosure(false);
+  const popoverId = useId();
+
+  return (
+    <>
+      <Box direction="row" alignItems="end">
+        <LakeLabel
+          label={label}
+          style={styles.label}
+          actions={
+            <LakeButton mode="secondary" icon="calendar-ltr-regular" size="small" onPress={open} />
+          }
+          render={id => (
+            <Rifm value={value ?? ""} onChange={onChange} {...rifmDateProps}>
+              {({ value, onChange }) => (
+                <LakeTextInput
+                  ref={ref}
+                  id={id}
+                  placeholder={format}
+                  value={value}
+                  error={error}
+                  onChange={onChange}
+                  ariaExpanded={isOpened}
+                />
+              )}
+            </Rifm>
+          )}
+        />
+      </Box>
+
+      <Popover id={popoverId} role="dialog" onDismiss={close} referenceRef={ref} visible={isOpened}>
+        <View style={desktop ? styles.popoverDesktop : styles.popover}>
+          <DatePickerPopoverContent
+            value={value}
+            format={format}
+            firstWeekDay={firstWeekDay}
+            monthNames={monthNames}
+            weekDayNames={weekDayNames}
+            desktop={desktop}
+            isSelectable={isSelectable}
+            onChange={onChange}
+          />
+        </View>
+      </Popover>
+    </>
+  );
+};
+
+type DatePickerPopoverProps = DatePickerProps & {
+  id?: string;
+  visible: boolean;
+  referenceRef: React.RefObject<unknown>;
+  cancelLabel: string;
+  confirmLabel: string;
+  onDissmiss: () => void;
+};
+
+export const DatePickerPopover = ({
+  value,
+  error,
+  format,
+  firstWeekDay,
+  monthNames,
+  weekDayNames,
+  isSelectable,
+  onChange,
+  id,
+  visible,
+  referenceRef,
+  label,
+  cancelLabel,
+  confirmLabel,
+  onDissmiss,
+}: DatePickerPopoverProps) => {
+  const { desktop } = useResponsive(DATE_PICKER_MOBILE_THRESHOLD);
+  const [localeValue, setLocaleValue] = useState(value);
+
+  useEffect(() => {
```

Comment:
Looks like an anti-pattern

Reply:
But in another PR this is replaced by using `useForm`

Reply:
Maybe you should merge the other PR in here.

Reply:
on one side it will make review of this component easier (only `DatePickerPopover` is impacted). But I don't want to make this PR even more bigger.
You can see changes here: https://github.com/swan-io/lake/pull/63/files#diff-8c4d9f8e6ad9ccd7d7325a533c549af7c97bd9edd0fe7b198a263317a339d491L883